solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBSolrClient.java

```diff
@@ -480,35 +488,70 @@ public RequestWriter getRequestWriter() {
+    // where another thread could mark it down, move it back to zombie, and then we delete
+    // from zombie and lose it forever.
+    ServerWrapper wrapper = zombieServers.remove(zombieServer.getBaseUrl());
+    if (wrapper != null) {
+      wrapper.failedPings = 0;
+      if (wrapper.standard) {
+        addToAlive(wrapper);
+      }
+    } else {
+      // something else already moved the server from zombie to alive
+    }
+  }
+
+  private void handleServerDown(ServerWrapper zombieServer) {
+    // Expected. The server is still down.
+    zombieServer.failedPings++;
+    zombieServer.skipAliveCheckIters = this.aliveCheckSkipIters;
+
+    // If the server doesn't belong in the standard set belonging to this load balancer
+    // then simply drop it after a certain number of failed pings.
+    if (!zombieServer.standard && zombieServer.failedPings >= NONSTANDARD_PING_LIMIT) {
+      zombieServers.remove(zombieServer.getBaseUrl());
+    }
+  }
+
   private void checkAZombieServer(ServerWrapper zombieServer) {
     try {
-      QueryRequest queryRequest = new QueryRequest(solrQuery);
-      queryRequest.setBasePath(zombieServer.baseUrl);
-      QueryResponse resp = queryRequest.process(getClient(zombieServer.getBaseUrl()));
-      if (resp.getStatus() == 0) {
-        // server has come back up.
-        // make sure to remove from zombies before adding to alive to avoid a race condition
-        // where another thread could mark it down, move it back to zombie, and then we delete
-        // from zombie and lose it forever.
-        ServerWrapper wrapper = zombieServers.remove(zombieServer.getBaseUrl());
-        if (wrapper != null) {
-          wrapper.failedPings = 0;
-          if (wrapper.standard) {
-            addToAlive(wrapper);
-          }
-        } else {
-          // something else already moved the server from zombie to alive
-        }
+      // push back on liveness checks only every Nth iteration
+      if (zombieServer.skipAliveCheckIters > 0) {
+        log.debug(
+            "Skipping liveness check for server "
+                + zombieServer.getBaseUrl()
+                + " because skipAliveCheckIters = "
+                + zombieServer.skipAliveCheckIters);
+        zombieServer.skipAliveCheckIters--;
+        return;
       }
-    } catch (Exception e) {
-      // Expected. The server is still down.
-      zombieServer.failedPings++;
 
-      // If the server doesn't belong in the standard set belonging to this load balancer
-      // then simply drop it after a certain number of failed pings.
-      if (!zombieServer.standard && zombieServer.failedPings >= NONSTANDARD_PING_LIMIT) {
-        zombieServers.remove(zombieServer.getBaseUrl());
+      if (isServerAlive(zombieServer)) {
+        log.debug("Successfully pinged server " + zombieServer.getBaseUrl() + ", marking it alive");
+        handleServerBackUp(zombieServer);
       }
+
+    } catch (Exception e) {
+      log.debug(
```

Comment:
and the check!   I think the whole check thing is werid that it's not just part of log.debug!